### DISCUSSION TITLE
COnsider let backgroundImage support Promise<ImageSource> ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [fabricjs/fabric.js#10221] [Ideas]
Does this need to wait until the image is loaded before new Canvas can be executed?
This kind of fragmentation behavior makes it difficult for users to package when using it

before
```
const imageELement = new Image()

new Canvas('',{
backgroundImage: imageELement
})
```

want
```
const imageELement = new Image()

new Canvas('',{
backgroundImage:FabricImage.loadImage('xxx.png') // support promise
})
```

**Comment by asturur** [comment]
You an already instantiate the canvas with a non loaded image and then load it yourself.
There is no difference between what you wrote and:

```
const image = await FabricImage.fromURL('xxx.png');
new Canvas('',{
backgroundImage: image
})
```

**Reply by vaynevayne** [reply]
ok，I want to do this,Need js constructor to support async, right? If you can't do it in parallel, is there a way to wait for pictures to load internally rather than when the user is using them?

**Reply by asturur** [reply]
a class instantiation can't be async. Not that i know of.
You are trying to solve an async issue and think the problem is in fabric.
Load your resources and then use them.

There is no need to load the pictures asyncronously internally.
You can do that already with fromObject if you really have to, but the problem is just in our you build your application flow.
